lib/rouge/lexers/supercollider.rb

```diff
@@ -34,39 +34,27 @@ def self.constants
         )
       end
 
-      start { push :bol }
-
-      # beginning of line
-      state :bol do
-        mixin :inline_whitespace
-
-        rule(//) { pop! }
-      end
-
-      state :inline_whitespace do
-        rule /\s+/m, Text
-        mixin :has_comments
-      end
-
       state :whitespace do
-        rule /\n+/m, Text, :bol
-        rule %r(\/\/\.*?$), Comment::Single, :bol
-        mixin :inline_whitespace
+        rule /\s+/m, Text
       end
 
-      state :has_comments do
-        rule %r(/[*]), Comment::Multiline, :nested_comment
+      state :comments do
+        rule %r(//.*?$), Comment::Single
+        rule %r(/[*]) do
+          token Comment::Multiline
+          push :nested_comment
+        end
       end
 
       state :nested_comment do
-        mixin :has_comments
+        rule %r(/[*]), Comment::Multiline, :nested_comment
         rule %r([*]/), Comment::Multiline, :pop!
         rule %r([^*/]+)m, Comment::Multiline
-        rule /./, Comment::Multiline
```

Comment:
This rule is likely necessary for things like `/* * */`, so that the middle `*` is not rendered as an error.

Reply:
Right you are!